src/System.Net.Ping/src/System/Net/NetworkInformation/Ping.Unix.cs

```diff
@@ -105,6 +106,17 @@ public partial class Ping
                     {
                         fixed (byte* bytesPtr = &receiveBuffer[0])
                         {
+                            if (isIpv4)
+                            {
+                                // Determine actual size of IP header
+                                byte ihl = (byte)(bytesPtr[0] & 0x0f); // Internet Header Length
+                                ipHeaderLength = 4 * ihl;
+                                if (bytesReceived - ipHeaderLength < IcmpHeaderLengthInBytes)
+                                {
+                                    continue; // Not enough bytes to reconstruct actual IP header + ICMP header.
```

Comment:
@mellinoe, I'm confused about the continue logic here.  It looks like we go back up and do another read, but we're reading back into receiveBuffer at position 0, and we don't seem to be tracking the total sum of everything that's been read, only what happened on the last read.  Am I reading that correctly?